fe/fe-core/src/main/java/com/starrocks/persist/gson/RuntimeTypeAdapterFactory.java

```diff
@@ -402,14 +379,8 @@ public R read(JsonReader in) throws IOException {
 
                 @SuppressWarnings("unchecked") // registration requires that subtype extends T
                 TypeAdapter<R> delegate = (TypeAdapter<R>) labelToDelegate.get(label);
-                if (delegate == null && fallbackLabel != null) {
-                    delegate = (TypeAdapter<R>) labelToDelegate.get(fallbackLabel);
-                    if (delegate != null) {
-                        LOG.warn("Fallback to deserialize object of label:{} to the label:{}", label, fallbackLabel);
-                    }
-                }
                 if (delegate == null) {
-                    throw new JsonParseException("cannot deserialize " + baseType + " subtype named " + label
+                    throw new SubtypeNotFoundException("cannot deserialize " + baseType + " subtype named " + label
```

Comment:
need to propagate the subtype name in the exception as well, so the upper level can reinterpret the error message.

Reply:
Yes, the error message is 
cannot deserialize class xxx subtype named xxx; did you forget to register a subtype?

Reply:
Do you mean to add the subType attribute in SubtypeNotFoundException?

Reply:
yes